gms/api/src/main/snapshot/com.linkedin.dashboard.dashboards.snapshot.json

```diff
@@ -0,0 +1,600 @@
+              "MANUAL" : "Manually provided by a user",
+              "OTHER" : "Other sources",
+              "SERVICE" : "Other ownership-like service, e.g. Nuage, ACL service etc",
+              "SOURCE_CONTROL" : "SCM system, e.g. GIT, SVN"
+            }
+          },
+          "doc" : "The type of the source"
+        }, {
+          "name" : "url",
+          "type" : "string",
+          "doc" : "A reference URL for the source",
+          "optional" : true
+        } ]
+      },
+      "doc" : "Source information for the ownership",
+      "optional" : true
+    } ]
+  }, {
+    "type" : "record",
+    "name" : "Ownership",
+    "namespace" : "com.linkedin.common",
+    "doc" : "Ownership information of an entity.",
+    "fields" : [ {
+      "name" : "owners",
+      "type" : {
+        "type" : "array",
+        "items" : "Owner"
+      },
+      "doc" : "List of owners of the entity."
+    }, {
+      "name" : "lastModified",
+      "type" : "AuditStamp",
+      "doc" : "Audit stamp containing who last modified the record and when."
+    } ]
+  }, "com.linkedin.common.OwnershipSource", "com.linkedin.common.OwnershipSourceType", "com.linkedin.common.OwnershipType", {
+    "type" : "record",
+    "name" : "Status",
+    "namespace" : "com.linkedin.common",
+    "doc" : "The status metadata of an entity, e.g. dataset, metric, feature, etc.",
+    "fields" : [ {
+      "name" : "removed",
+      "type" : "boolean",
+      "doc" : "whether the entity is removed or not",
+      "default" : false
+    } ]
+  }, "com.linkedin.common.Time", "com.linkedin.common.Urn", {
+    "type" : "record",
+    "name" : "Dashboard",
+    "namespace" : "com.linkedin.dashboard",
+    "doc" : "Metadata for a dashboard",
+    "include" : [ {
+      "type" : "record",
+      "name" : "DashboardKey",
+      "doc" : "Key for dashboard resource",
+      "fields" : [ {
+        "name" : "tool",
+        "type" : "string",
+        "doc" : "The name of the dashboard tool such as looker, redash etc.",
+        "validate" : {
+          "strlen" : {
+            "max" : 20,
+            "min" : 1
+          }
+        }
+      }, {
+        "name" : "dashboardId",
+        "type" : "string",
+        "doc" : "Unique id of the dashboard assigned by the specific dashboard tool",
+        "validate" : {
+          "strlen" : {
+            "max" : 500,
+            "min" : 1
+          }
+        }
+      } ]
+    } ],
+    "fields" : [ {
+      "name" : "info",
+      "type" : {
+        "type" : "record",
+        "name" : "DashboardInfo",
+        "doc" : "Information about a dashboard",
+        "fields" : [ {
+          "name" : "title",
+          "type" : "string",
+          "doc" : "Title of the dashboard"
+        }, {
+          "name" : "description",
+          "type" : "string",
+          "doc" : "Detailed description about the dashboard"
+        }, {
+          "name" : "charts",
+          "type" : {
+            "type" : "array",
```

Comment:
How does Looker fit into the "a dashboard has multiple charts" modeling. 

Looker lets you do a couple of different things:
- Add a look to a dashboard as a tile.
- Add an explore (with certain filters applied) to a dashboard as a tile. 

Is the idea that an "explore" with certain filters applied would be modeled as a "chart" and a "look" would be modeled as a chart? As I mentioned above, there can be some funkiness because Looker treats explores and looks differently.

Reply:
Again, appreciate the expert review. Probably I wasn't quite careful in my study to explore "explore"s in Looker :) Do you think it should be a separate entity? Or, is it basically another look with certain filters applied?

Reply:
"Explores" are kind of like views in Looker, they represent virtual tables with pre-joined relationships between the dependent tables so people can click around the Looker UI filters to generate queries. 
https://docs.looker.com/reference/explore-params/explore

As I mentioned above, Looker lets you add charts to a dashboard in two different ways:
- Add a look to a dashboard. 
- Add an explore with certain filters applied to a dashboard. This is essentially like a "Look" but it is unnamed (while a Look is a named entity). It also has a different URI structure. 

Look URI:
` ${BASE_URI}/looks/${LOOK_ID}`

Explore URI:
`${BASE_URI}/explore/${explore_name}/history?qid=${unique_query_id_hash}`

Reply:
Also if it helps, here is a public Looker instance you can play around with that contains covid-19 data. Might help to understand the Looker concepts:
https://covid19response.cloud.looker.com/embed/dashboards-next/51

Reply:
Thanks for the pointers. I started to think that looks and explores are two different customizations of "chart". Your thinking is right, to be able to differentiate between looks vs explores, part of URI could be put into the URN definition. For example: chartId could be `looks/${LOOK_ID}` for looks and `explore/${explore_name}` for explores.

On the other hand, there is another way to approach this problem. Modelling dashboard metadata was a little tricky because we thought that the models provided should be easily applicable/extensible to any dashboarding tool out there. So, we came up with two entities which are dashboard and chart. These are most basic building blocks for a dashboarding platform. If Looker has an additional entity (I guess the concept of `board` could be good example), extra modelling will be needed for that. Or if Looker has two very different types of charts, extra modelling *might* be needed like `TypeAChart extends Chart` and `TypeBChart extends Chart`. One thing to be careful here, Pegasus doesn't quite have inheritance but it has composition: https://linkedin.github.io/rest.li/pdl_schema#including-fields